src/Workspaces/CSharp/Portable/Utilities/SpeculationAnalyzer.cs

```diff
@@ -487,6 +495,26 @@ protected override bool ReplacementChangesSemanticsForNodeLanguageSpecific(Synta
             return false;
         }
 
+        /// <summary>
+        /// Checks if the conversion might change the resultant boxed type.
+        /// </summary>
+        private bool ReplacementBreaksBoxingInConditionalExpression(TypeInfo originalExpressionTypeInfo, TypeInfo newExpressionTypeInfo, ExpressionSyntax previousOriginalNode, ExpressionSyntax previousReplacedNode)
+        {
+            // If the resultant types are different and it is boxing to the converted type then semantics could be changing.
+            if (originalExpressionTypeInfo.Type != newExpressionTypeInfo.Type)
```

Comment:
don't use `!=` on types.  use `!Equals(type1, type2)`.  Types are not guaranteed to have reference equality.

Reply:
Thanks for the tip - so there's actually a couple other places in this class that use == instead of Equals.

http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/Utilities/SpeculationAnalyzer.cs,249

http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/Utilities/SpeculationAnalyzer.cs,400

Unless you see an issue, I'll go ahead and change those to see if tests start breaking.

Reply:
I also filed https://github.com/dotnet/roslyn-analyzers/issues/2084 - this happens way too often to not have an analyzer/fixer for it.